### PR TITLE
SSCS-9668

### PR DESCRIPTION
**Pull request by gmikerlemos**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-9668

### Change description ###

Child Support - Upload response event - edited reason child support confidentiality

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/createbundle/CreateBundleAboutToSubmitHandler.java [hunk 1]
```diff
@@ -199,6 +200,10 @@ private boolean hasPhmeRequestOrConfidentialityUnderReview(SscsCaseData sscsCase
         return !response.getErrors().isEmpty();
     }
 
+    private boolean isPhmeAbleBenefitTypeCase(SscsCaseData sscsCaseData) {
+        return !sscsCaseData.isBenefitType(Benefit.CHILD_SUPPORT);
```
Comment (jackmaloney) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/createbundle/CreateBundleAboutToSubmitHandler.java:204]: The method name of this one doesn't really make sense to me considering it's just checking if it is not a child support case. I've tried to get my head round this whole logic a few times and im pretty confused as to what is going on. Is there anyway to make clearer? Or if not, maybe put a comment as to what this is actually doing?

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/dwpuploadresponse/DwpUploadResponseAboutToSubmitHandlerTest.java [hunk 2]
```diff
@@ -821,6 +821,76 @@ public void ucbEvidenceDocument_mustBeAPdf() {
...
+
+        PreSubmitCallbackResponse<SscsCaseData> response = dwpUploadResponseAboutToSubmitHandler
+                .handle(ABOUT_TO_SUBMIT, callback, USER_AUTHORISATION);
+
+        assertThat(response.getErrors().iterator().next(),
+                is("Child support - Confidentiality is not a valid selection for this case"));
+    }
+
+    @Test
+    public void givenADwpUploadResponseEventPmheNonChildSupportThenNoErrorAdded() {
```
Comment (jackmaloney) [inline src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/dwpuploadresponse/DwpUploadResponseAboutToSubmitHandlerTest.java:861]: typo in test name for phme

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/createbundle/CreateBundleAboutToSubmitHandlerTest.java [hunk 3]
```diff
@@ -242,6 +242,19 @@ public void givenCaseWithEditedDwpDocsAndPhmeUnderReview_thenReturnErrorMessageA
...
+    public void givenCaseWithEditedDwpDocsAndChildSupport_thenReturnNoError() {
+        addMandatoryDwpDocuments();
+        SscsCaseData sscsCaseData = callback.getCaseDetails().getCaseData();
+        sscsCaseData.setLanguagePreferenceWelsh(NO.getValue());
+        sscsCaseData.getAppeal().setBenefitType(BenefitType.builder().code("childSupport").build());
+        sscsCaseData.setBenefitCode("022");
+
+        PreSubmitCallbackResponse<SscsCaseData> response = handler.handle(ABOUT_TO_SUBMIT, callback, USER_AUTHORISATION);
+
+        assertEquals(0, response.getErrors().size());
```
Comment (jackmaloney) [inline src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/createbundle/CreateBundleAboutToSubmitHandlerTest.java:255]: Is this actually a useful test? Shouldn't we check that the class actually hits the expected methods and gets the correct number of bundles when it is a Child Support case with edited docs?

**Review by jackmaloney — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/dwpuploadresponse/DwpUploadResponseAboutToSubmitHandler.java [hunk 4]
```diff
@@ -93,7 +93,7 @@ protected void handleAudioVideoDocuments(SscsCaseData sscsCaseData) {
 
         List<AudioVideoEvidence> dwpAudioVideoEvidence = sscsCaseData.getDwpUploadAudioVideoEvidence();
 
-        for (AudioVideoEvidence audioVideo: dwpAudioVideoEvidence) {
+        for (AudioVideoEvidence audioVideo : dwpAudioVideoEvidence) {
```
[the remark this answers is not in the document]
Comment (gmikerlemos) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/dwpuploadresponse/DwpUploadResponseAboutToSubmitHandler.java:96]: ???

**Review by jackmaloney — APPROVED** [review]

**Pull request merged**